drivers/storage/port/scsiport/fdo.c

```diff
@@ -732,6 +777,20 @@ FdoDispatchPnp(
         {
             return FdoHandleDeviceRelations(portExt, Irp);
         }
+        case IRP_MN_QUERY_ID:
+        {
+            if (ioStack->Parameters.QueryId.IdType == BusQueryCompatibleIDs)
+            {
+                status = IoForwardIrpSynchronously(portExt->Common.LowerDevice, Irp);
+                if (!NT_SUCCESS(status))
+                {
+                    Irp->IoStatus.Information = 0;
+                }
```

Comment:
```suggestion
                Irp->IoStatus.Information = 0;
                IoForwardIrpSynchronously(portExt->Common.LowerDevice, Irp);
```

Reply:
You don't look at the returned value of IoForwardIrpSynchronously?

Reply:
Yep, we continue execution (appending the ID) regardless of it.